In [28]:
# Import packages
import os
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import math
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.linear_model import LinearRegression,LogisticRegression
from sklearn.model_selection import KFold,cross_val_score
%matplotlib inline

In [29]:
f=open('Training.txt','r',encoding='UTF-8')
training = f.read().replace(u'\ufeff','').splitlines()
training_set=[training[i].split(',') for i in range (len(training))]
for i in range (len(training_set)):
    for j in range (513):
        training_set[i][j]=float(training_set[i][j])
trn=pd.DataFrame(training_set)
Xtrn=trn.loc[:,0:511]
Ytrn=trn[[512]]

In [30]:
f=open('Testing.txt','r',encoding='UTF-8')
testing = f.read().replace(u'\ufeff','').splitlines()
testing_set=[testing[i].split(',') for i in range (len(testing))]
for i in range (len(testing_set)):
    for j in range (513):
        testing_set[i][j]=float(testing_set[i][j])
tst=pd.DataFrame(testing_set)
Xtst=tst.loc[:,0:511]
Ytst=tst[[512]]

Since we only have two classes in this problem, logistic regression would be a suitable model. 

In [31]:
lg=LogisticRegression()
lg.fit(Xtrn,np.ravel(Ytrn))

LogisticRegression()

In [32]:
lg.score(Xtst,Ytst)

0.6690647482014388

In [33]:
cm=confusion_matrix(Ytst,lg.predict(Xtst))
cm

array([[87, 17],
       [29,  6]], dtype=int64)

The accuracy of predictions on test data is about 66.9%. However, since we want to predict as many major earthquakes as possible, we are more interested in the recall. According to the confusion matrix, the recall is 6/(6+29)=17%, which is very poor. 

One of the reasons is that there are many other geographical features that affect the earthquake, for example, the distance to boundaries between techtonic plates. Such data may also be included to predict major earthquake events.